content/_changelogs/12.3.0.md

```diff
@@ -0,0 +1,81 @@
+## 12.3.0
+
+_Released 1/03/2023_
+
+**Features:**
+
+- Added support for mapping the `CYPRESS_PULL_REQUEST_ID`,
+  `CYPRESS_PULL_REQUEST_URL`, and/or `CYPRESS_CI_BUILD_URL` environment
+  variables to the recorded Cloud run. This provides workarounds when supported
+  CI provider mappings are incorrect or unsupported CI providers are used.
+  Addressed in [#25036](https://github.com/cypress-io/cypress/pull/25036).
+- Added new Cypress API, [`Cypress.currentRetry`](/api/api/currentretry), to
+  easily access the current test retry count. Addresses
+  [#25239](https://github.com/cypress-io/cypress/pull/25239).
+
+**Performance:**
+
+- Increased the pre-request proxy cleanup interval to 10s. The previous cleanup
+  interval was too aggressed for projects loading a large number of JS modules,
+  causing applications to load very slowly and in chunks. Fixed in
+  [#25209](https://github.com/cypress-io/cypress/pull/25209).
+- Fixed an issue where browsers distributed as universal binaries (Chrome,
+  Firefox) on M1 Macs could be launched in the wrong architecture, resulting in
+  poor performance in-browser. Fixed in
+  [#25014](https://github.com/cypress-io/cypress/pull/25014).
+- Resolved long startup delays when running Cypress within a Cypress-provided
+  docker image when the project uses a custom cypress configuration file
```

Comment:
This fix wasn't specific to Docker images, but would be most often observed in that use case since Docker is slower. Maybe something like this?

"Resolved a delay that could occur on startup when using a custom Cypress configuration file location in projects with a transitive `typescript` dependency."